TriggerRunnerSupervisor is flaky

Happens intermittently:

```
FAILED airflow-core/tests/unit/jobs/test_triggerer_job.py::test_trigger_lifecycle - Failed: TriggerRunnerSupervisor never deleted trigger
```

```python
________________________________________________________________________________________ test_trigger_lifecycle _________________________________________________________________________________________
airflow-core/tests/unit/jobs/test_triggerer_job.py:252: in test_trigger_lifecycle
    pytest.fail("TriggerRunnerSupervisor never deleted trigger")
E   Failed: TriggerRunnerSupervisor never deleted trigger
------------------------------------------------------------------------------------------ Captured stdout call ------------------------------------------------------------------------------------------
[2025-08-07T14:45:57.554+0000] {signature.py:556} WARNING - Determined the owner of <function WatchedSubprocess.send_msg at 0x7fc735895080> to be <class 'airflow.sdk.execution_time.supervisor.WatchedSubprocess'>, but it may be wrong. Please pass owner= to spy_on() to set a specific owner.
2025-08-07 14:45:58 [warning  ] The `airflow.utils.timezone.convert_to_utc` attribute is deprecated. Please use `'airflow.sdk.timezone.convert_to_utc'`. [py.warnings] [airflow.jobs.triggerer_job_runner] category=DeprecationWarning filename=/opt/airflow/providers/standard/src/airflow/providers/standard/triggers/temporal.py lineno=51
2025-08-07 14:46:00 [info     ] Triggerer's async thread was blocked for 0.79 seconds, likely by a badly-written trigger. Set PYTHONASYNCIODEBUG=1 to get more information on overrunning coroutines. [airflow.jobs.triggerer_job_runner]
2025-08-07 14:46:06 [info     ] Process exited                 [supervisor] exit_code=<Negsignal.SIGINT: -2> pid=110 signal_sent=SIGINT
````


For example: https://github.com/apache/airflow/actions/runs/16806224260/job/47604317411

--- a/airflow-core/tests/unit/jobs/test_triggerer_job.py
+++ b/airflow-core/tests/unit/jobs/test_triggerer_job.py
@@ -264,8 +264,8 @@ def test_trigger_lifecycle(spy_agency: SpyAgency, session, testing_dag_bundle):
         # Re-load the triggers
         trigger_runner_supervisor.load_triggers()
 
-        # Wait for up to 3 seconds for it to vanish from the TriggerRunner's storage
-        for _ in range(30):
+        # Wait for up to 10 seconds for it to vanish from the TriggerRunner's storage
+        for _ in range(100):
             if not trigger_runner_supervisor.running_triggers:
                 break
             trigger_runner_supervisor._service_subprocess(0.1)

